src/Hosting/Hosting/test/Internal/HostTests.AsyncDisposable.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if NETCOREAPP
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.Extensions.Hosting.Internal
+{
+    public partial class HostTests
+    {
+        [Fact]
+        public async Task HostCallsDisposeAsyncOnServiceProvider()
```

Comment:
No test that directly calls DisposeAsync?

Reply:
Added